pkg/apis/storage/types.go

```diff
@@ -157,7 +157,12 @@ type VolumeAttachmentSource struct {
 	// +optional
 	PersistentVolumeName *string
 
-	// Placeholder for *VolumeSource to accommodate inline volumes in pods.
+	// Copy of VolumeSource to accommodate VolumeSource from a pod
+	// to support shimming of in-tree inline volumes to a CSI backend.
+	// This field is alpha-level and is only honored by servers that
+	// enable the CSIMigration feature.
+	// +optional
+	InlineVolumeSource *api.VolumeSource
```

Comment:
some volume sources contain object references which need to refer to things in the source pod's namespace (e.g. ISCSI.SecretRef, Glusterfs.EndpointsName, RBD.SecretRef, Cinder.SecretRef, etc). how is the originating namespace tracked so those can be resolved correctly?

Reply:
ooh, great point

Reply:
Maybe this should be a Pod reference with namespace and name plus a volume name within that pod?  There's security implications there, I guess.  Maybe simply copying namespace here is enough.

Reply:
This has been resolved by making the `InlineVolumeSource` a `CSIPersistentVolumeSource`. Since `CSIPersistentVolumeSource` is the "end result" that the external-attacher uses to attach the volumes it contains [and must always contain] all the requisite information for attach including secrets and secret namespaces. That then pushes the onus of translating the original inline volume to a `CSIPersistentVolumeSource` to the AttachDetachController which should have all the required information to populate the new field.